### DISCUSSION TITLE
Auth token invalid after `confirmEmailChange`

### DISCUSSION THREAD

**Discussion by Amir-Zouerami** [pocketbase/pocketbase#3828] [Q&A] [answered]
I couldn't find any details about this in [Web APIs reference](https://pocketbase.io/docs/api-records#confirm-email-change) after a glance, but is the token invalidated after `confirmEmailChange`? In my testing after successfully calling `confirmEmailChange` with the token and password, I don't seem to be able to even run `authRefresh` (I get 401 unauthorized). Do I have to fully re-authenticate the user? The `isValid` property returns true by the way but that's because the `authStore` is not updated. I get this in the logs after `authRefresh`:

```
{
  "errorDetails": "<nil>",
  "errorMessage": "The request requires valid record authorization token to be set."
}
```

If I want to do that (re-authentication) on user's behalf, I need to have the new email that is confirmed. How do I get that considering the model is not updated automatically and I can't call `authRefresh`? I ask it directly from the user alongside the password in a form for now, but I'm open to better suggestions.

**Comment by ganigeorgiev** [comment, marked as answer]
Yes, currently after successful `confirmEmailChange` action the previously issued tokens will be invalidated (_this behavior for the confirm email change action may change in the future as there are plans to slightly rework the auth flow to fix some implementation warts regarding OAuth2 providers that doesn't return an email_).

To reauthenticate you can extract the `newEmail` field from the JWT payload, something like:

```js
import PocketBase, { getTokenPayload } from "pocketbase"

// the token can be retrieved from the url with example using URLSearchParams
const newEmail = getTokenPayload(token).newEmail;

...
```

**Reply by Amir-Zouerami** [reply]
That's very useful. It would be nice if confirmEmailChange returns the new email in the future though!
